samza-api/src/main/java/org/apache/samza/metadatastore/MetadataStore.java

```diff
@@ -49,6 +49,17 @@
    */
   void put(String key, byte[] value);
 
+  /**
+   * Updates the mapping with the specified map.
+   *
+   * @param entries mapping of key to values to write to the metadata store
+   */
+  default void putAll(Map<String, byte[]> entries) {
```

Comment:
Can you please add documentation around the new API? Especially around no guarantees on atomicity of the writes. If you'd like to address the documentation as part of a follow PR for the all the APIs, that is fine by me too.

Reply:
No problem, will add.